presto-main/src/main/java/com/facebook/presto/sql/analyzer/StatementAnalyzer.java

```diff
@@ -344,6 +357,79 @@ protected RelationType visitShowColumns(ShowColumns showColumns, AnalysisContext
     }
 
     @Override
+    protected RelationType visitShowCreateTable(ShowCreateTable showCreateTable, AnalysisContext context)
+    {
+        QualifiedObjectName tableName = createQualifiedObjectName(session, showCreateTable, showCreateTable.getName());
+        Optional<TableHandle> tableHandle = metadata.getTableHandle(session, tableName);
+
+        if (metadata.getView(session, tableName).isPresent()) {
+            throw new SemanticException(NOT_SUPPORTED, showCreateTable, "'%s' is a view, not a table", tableName);
+        }
+
+        if (!tableHandle.isPresent()) {
+            throw new SemanticException(MISSING_TABLE, showCreateTable, "Table '%s' does not exist", tableName);
+        }
+
+        ConnectorTableMetadata connectorTableMetadata = metadata.getTableMetadata(session, tableHandle.get()).getMetadata();
+
+        List<TableElement> columns = connectorTableMetadata.getColumns().stream()
+                .filter(column -> !column.isHidden())
+                .map(column -> new TableElement(column.getName(), column.getType().getDisplayName()))
+                .collect(toImmutableList());
+
+        Map<String, Object> properties = connectorTableMetadata.getProperties();
+        Map<String, PropertyMetadata<?>> allTableProperties = metadata.getTablePropertyManager().getAllTableProperties().get(tableName.getCatalogName());
+        Map<String, Expression> sqlProperties = new HashMap<>();
+
+        for (Map.Entry<String, Object> propertyEntry : properties.entrySet()) {
+            String propertyName = propertyEntry.getKey();
+            Expression sqlExpression = getExpression(allTableProperties.get(propertyName)::encode, propertyEntry.getValue());
+            sqlProperties.put(propertyName, sqlExpression);
+        }
+
+        CreateTable createTable = new CreateTable(QualifiedName.of(tableName.getCatalogName(), tableName.getSchemaName(), tableName.getObjectName()), columns, false, sqlProperties);
+        Query query = singleValueQuery("Create Table", SqlFormatter.formatSql(createTable));
+
+        return process(query, context);
+    }
+
+    private static <T> Expression getExpression(Function<T, Object> function, Object object)
+    {
+        requireNonNull(object, "Property value cannot be null");
+        Object encoded = function.apply((T) object);
+        return parseExpression(encoded);
+    }
+
+    private static Expression parseExpression(Object object)
+    {
+        if (object instanceof String) {
+            return new StringLiteral(object.toString());
+        }
+
+        if (object instanceof Boolean) {
+            return new BooleanLiteral(object.toString());
+        }
+
+        if (object instanceof Long || object instanceof Integer) {
+            return new LongLiteral(object.toString());
+        }
+
+        if (object instanceof Double) {
+            return new DoubleLiteral(object.toString());
+        }
+
+        if (object instanceof List) {
+            List list = (List) object;
```

Comment:
The List does not contain Expressions, they are the String, Boolean, etc types. They need to go through StatementAnalyzer::toExpression to become Expression.